client/src/__tests__/AppRoutes.test.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { configure, mount } from 'enzyme';
+import { Redirect, MemoryRouter } from 'react-router';
+import AppRoutes from '../components/routes';
+import Index from '../components/index/index';
+import Create from '../components/createBlog/create';
+import Adapter from 'enzyme-adapter-react-16';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+
+configure({ adapter: new Adapter() });
+const initialState = { auth: { isLoggedIn: true } }
+
+describe('Routes tests', () => {
+  const mockStore = configureStore();
+  let store, wrapper;
+
+  beforeEach(() => {
+    store = mockStore(initialState)
+  })
+
+  it('Index route', () => {
+
+    wrapper = mount(
+      <MemoryRouter initialEntries={['/whatever']}>
+        <AppRoutes store={store} />
+      </MemoryRouter>
+    );
+    expect(wrapper.find(Index)).toHaveLength(1);
```

Comment:
For testing react's component (render()) best practice use shapshot way.

`expect(wrapper).toMatchShapshot()`;

Read more about this, please

Reply:
I found out this practice better for combination react router + jest + enzyme